### PR TITLE
Closes #67

### LINKED ISSUE #67: clean up the 2 methods for `McmcOptions`
See https://github.com/Roche/crmPack/blob/main/R/McmcOptions-methods.R

To do:

- [x] Revise documentation
  - [ ] RStudio outline section dividers (use `# function or class name ----`)
  - [ ] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [ ] Replace `##'` by `#'`
  - [ ] Add to `pkgdown.yml`
  - [ ] Remove file history and the timestamp in comments on top of file
- [ ] Add unit tests
  - [ ] For classes:
    - [ ] Test default constructor
    - [ ] For reference classes: test methods
  - [ ] For functions: as usual
- [ ] Refactor as needed
  - [ ] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [ ] Keep externally used names so that we don't break downstream code
  - [ ] Remove `keywords` tags
  - [ ] For functions or methods:
    - [ ] Add assertions for arguments using `checkmate` functions
    - [ ] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [ ] Consider renaming internal variables etc. if needed
    - [ ] Avoid explicit use of `return`
  - [ ] For classes:
    - [ ] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [ ] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [ ] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #67

### CODE REVIEW

**Comment by wwojciech** [comment]
Note: #73

**Comment by wwojciech** [comment]
Hi @dinakar29 would you be having any idea on why this automation check fails at `Docs` phase? Tx!

**Comment by dinakar29** [comment]
> Hi @dinakar29 would you be having any idea on why this automation check fails at `Docs` phase? Tx!

Currently looking into this issue. Will let you know following a resolution.

File: NAMESPACE [hunk 1]
```diff
@@ -106,8 +106,8 @@ export(plotDualResponses)
 export(plotGain)
 export(prob)
 export(probit)
-export(sampleSize)
-export(saveSample)
```
Comment (danielinteractive) [inline NAMESPACE:110]: please keep the original `camelCase` way of the function names (as mentioned before), as the whole package is like that and I don't want to change too much. thx

File: R/McmcOptions-methods.R [hunk 2]
```diff
@@ -1,51 +1,78 @@
...
+# McmcOptions-sample_size ----
+
+#' @rdname sample_size
+#' @aliases sample_size-McmcOptions-method
+#' @example examples/McmcOptions-methods-sample_size.R
+setMethod(
+  f = "sample_size",
+  signature = signature(object = "McmcOptions"),
+  definition = function(object, ...) {
+    # Iteration numbers relative to object@burnin.
```
Comment (danielinteractive) [inline R/McmcOptions-methods.R:35]:
```suggestion
```
(clear from code)

File: R/McmcOptions-methods.R [hunk 3]
```diff
@@ -1,51 +1,78 @@
...
+# McmcOptions-save_sample ----
+
+#' @rdname save_sample
+#' @aliases save_sample-McmcOptions-method
+#' @example examples/McmcOptions-methods-save_sample.R
+setMethod(
+  f = "save_sample",
+  signature = signature(object = "McmcOptions"),
+  definition = function(object, iteration, ...) {
+    iteration_relative <- iteration - object@burnin # Relative to object@burnin.
```
Comment (danielinteractive) [inline R/McmcOptions-methods.R:75]:
```suggestion
    iteration_relative <- iteration - object@burnin
```

**Review by danielinteractive — COMMENTED** [review]
thx Wojciech, please generally keep function names as before - you can still change / clean up names of internal variables within functions of course. thx

**Review by danielinteractive — APPROVED** [review]

**Comment by dinakar29** [comment]
Can't figure out why I cannot reproduce this locally on the same container, but it doesn't work the same on the GH action workflow.

For now, please ignore the pkgdown workflow results until I investigate this further and to unblock progress on other work.

**Comment by danielinteractive** [comment]
Thanks @dinakar29 for looking into it, have a good holiday break first now !

**Comment by wwojciech** [comment]
Thank you Dinakar! @dinakar29 Happy Christmas!

**Pull request merged**